modules/docker-build/main.tf

```
@@ -41,3 +41,10 @@ resource "aws_ecr_repository" "this" {
 
   tags = var.ecr_repo_tags
 }
+
+resource "aws_ecr_lifecycle_policy" "this" {
+  count = length(var.ecr_repo_lifecycle_policy) > 0 ? 1 : 0
```

`var.ecr_repo_lifecycle_policy` is not a list, but a string (should be `null` by default).

```suggestion
  count = var.ecr_repo_lifecycle_policy != null ? 1 : 0
```